src/app/podcasts/[id]/page.tsx

```diff
@@ -0,0 +1,113 @@
+"use client"
+import axios from "axios"
+import { useEffect, useState } from "react"
+import { usePathname, useSearchParams } from "next/navigation"
+import { useRouter } from "next/navigation"
+import Image from "next/image"
+import SpotifyLogo from "/public/images/logos/spotify-logo.svg"
+
+import { PodcastEpisode } from "../../../../utils/static/fetchPodcastsFromSpotify"
+
+const EpisodePage = () => {
+  const router = useRouter()
+  const pathname = usePathname()
+  const searchParams = useSearchParams()
+
+  const idSegment = pathname.split("/").pop()
+  const id = idSegment || searchParams.get("id") || ""
+
+  const [episode, setEpisode] = useState<PodcastEpisode | null>(null)
+
+  useEffect(() => {
+    const fetchEpisode = async () => {
+      try {
+        if (id) {
+          const response = await axios.get(
+            `https://spotify23.p.rapidapi.com/episode/?id=${id}`,
+            {
+              headers: {
+                "X-RapidAPI-Key":
+                  "ac41cab2aamsh587d2717936bec9p185b22jsna20c36b1f5ef",
+                "X-RapidAPI-Host": "spotify23.p.rapidapi.com",
+              },
```

Comment:
you can extract these headers to `constants.ts` so you can reuse it again and again